### PR TITLE
added PublicAPI class and etc

### PR DESCRIPTION
**Pull request by Allerter**
This PR makes the following additions/changes:
- Adds the PublicAPI class which covers the public endpoints available at genius.com/api. I tried to find as many of these endpoints as possible and to make sense of their parameters and responses. But since they're undocumented, can't be 100 percent sure.
- renamed all the methods with a `get_` as an experiment to see how they'd look without that. But if you think it's better to leave the `get_` part be, I'll easily revert it back.
- As there are now two API classes each with its own _make_request, I created a `Sender` class to be their base. Also, the `API` and PublicAPI` class can be initialized on their own like before, and `PublicAPI` doesn't need a token.
- renamed `_scrach_lyrics_from_url` to `lyrics` to allow users to use this method by passing either song URLs or song IDs.
- reformatted the `Artist` object to allow for a `client` attribute. With this change, now the `artist.song` (formerly `artist.get_song`) is implemented (now it only gets a song and doesn't add it) and `artist.add_song` supports song names beside Song objects.
- updated some of the dependencies' versions and changed `.travis.yml` and `tox.ini` to allow for all tests (unit tests, code style tests and etc) on Travis CI.

### CODE REVIEW

**Review by johnwmillr — APPROVED** [review]

File: lyricsgenius/api/base.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import time
 
 
-class Sender:
+class Sender(object):
```
Comment (Allerter) [inline lyricsgenius/api/base.py:6]: All base classes inherit `object` in Python 3 regardless of typing `(object)` or not. So I left it out.

File: lyricsgenius/artist.py [hunk 2]
```diff
@@ -18,9 +18,10 @@ class Artist(object):
...
     """
 
-    def __init__(self, json_dict):
+    def __init__(self, client, json_dict):
         # Artist Constructor
 
+        self._client = client
         self._body = json_dict['artist']
         self._url = self._body['url']
         self._api_path = self._body['api_path']
```
Comment (johnwmillr) [inline lyricsgenius/artist.py:27]: Eventually we should add documentation for the inputs to the `Artist` class.

**Review by johnwmillr — APPROVED** [review]
This a great, PR! Thank you @Allerter, for adding in support for the Public API.

**Comment by Allerter** [comment]
Thanks for reviewing and merging the PR. A question: why not squash and merge?

**Comment by johnwmillr** [comment]
Just a total mistake on my end, I'm afraid

**Pull request merged**